### ISSUE TITLE
Can I have two SDP(Single Date Pickers) in the same component?

### ISSUE THREAD

**Issue by diegowifi** [airbnb/react-dates#983]
I want to use an individual date picker for the **start date** and the second for an **end date** 

This is my code, as you can see I'm using the same code as in the docs, I think the issue is with the state because they have the same twice.

```
<SingleDatePicker
  date={this.state.date} // momentPropTypes.momentObj or null
  onDateChange={date => this.setState({ date })} // PropTypes.func.isRequired
  focused={this.state.focused} // PropTypes.bool
  onFocusChange={({ focused }) => this.setState({ focused })} // PropTypes.func.isRequired
/>

<SingleDatePicker
  date={this.state.date} // momentPropTypes.momentObj or null
  onDateChange={date => this.setState({ date })} // PropTypes.func.isRequired
  focused={this.state.focused} // PropTypes.bool
  onFocusChange={({ focused }) => this.setState({ focused })} // PropTypes.func.isRequired
/>
```

**Comment by ljharb** [comment]
Are you storing both `date`s and `focused`s in your outer component's state under the same name?

**Comment by diegowifi** [comment]
Yes @ljharb , here is the full code of my component, I just got started working with React so maybe the issue is quite obvious but I can´t see it. Both datepickers are not working at this point.

```
import React from "react";
import moment from "moment";
import 'react-dates/initialize';
import { DateRangePicker, SingleDatePicker, DayPickerRangeController } from 'react-dates';
import 'react-dates/lib/css/_datepicker.css';

moment.locale('es');

class NewEvent extends React.Component {

  constructor(props) {
    super(props);

    this.state = {
      today: moment().format('DD'+'/'+'MM'+'/'+'YYYY'),
      date: null,
    };
  }

  render() {

    return(
      <form>
        <SingleDatePicker
          id="first"
          date={this.state.date} // momentPropTypes.momentObj or null
          onDateChange={date => this.setState({ date })} // PropTypes.func.isRequired
          focused={this.state.focused} // PropTypes.bool
          onFocusChange={({ focused }) => this.setState({ focused })} // PropTypes.func.isRequired
          enableOutsideDays={false}
          displayFormat="DD/MM/YYYY"
          placeholder={this.state.today}
        />
        <SingleDatePicker
          id="second"
          date={this.state.date} // momentPropTypes.momentObj or null
          onDateChange={date => this.setState({ date })} // PropTypes.func.isRequired
          focused={this.state.focused} // PropTypes.bool
          onFocusChange={({ focused }) => this.setState({ focused })} // PropTypes.func.isRequired
          enableOutsideDays={false}
          displayFormat="DD/MM/YYYY"
          placeholder={this.state.today}
        />
      </form>
    )
  }
}

export default NewEvent;

```

**Comment by diegowifi** [comment]
Like this the first SDP works but the second doesn´t. 

Is this what you meant by changing the state?

```
class NewEvent extends React.Component {

  constructor(props) {
    super(props);

    this.state = {
      date: null,
      dateTwo: null,
      focusedTwo: false,
    };
  }

  render() {

    return(
      <form>
        <SingleDatePicker
          id="first"
          date={this.state.date} // momentPropTypes.momentObj or null
          onDateChange={date => this.setState({ date })} // PropTypes.func.isRequired
          focused={this.state.focused} // PropTypes.bool
          onFocusChange={({ focused }) => this.setState({ focused })} // PropTypes.func.isRequired
        />
        <SingleDatePicker
          id="second"
          date={this.state.dateTwo} // momentPropTypes.momentObj or null
          onDateChange={date => this.setState({ date })} // PropTypes.func.isRequired
          focused={this.state.focusedTwo} // PropTypes.bool
          onFocusChange={({ focused }) => this.setState({ focused })} // PropTypes.func.isRequired
        />
      </form>
    )
  }
}

export default NewEvent;
```

**Comment by ljharb** [comment]
`onFocusChange={({ focused }) => this.setState({ focused })}` is setting a state called "focused" twice, and `onDateChange={date => this.setState({ date })}` is setting a state called "date" twice. At least one of those pairs needs to be renamed.

**Comment by diegowifi** [comment]
Thanks for all the feedback @ljharb, so now both SDPs work but the second one is always showing up because I set the state to `focusedTwo: true`. I need to handle the focus right.

I think I'm not passing something right because `onFocusChange` needs a `PropTypes.func.isRequired` and I don't know what's that.

```
class NewEvent extends React.Component {

  constructor(props) {
    super(props);

    this.state = {
      date: null,
      dateTwo: null,
      focusedTwo: true
    };
  }

  render() {

    return(
      <form>
        <SingleDatePicker
          date={this.state.date} // momentPropTypes.momentObj or null
          onDateChange={date => this.setState({ date })} // PropTypes.func.isRequired
          focused={this.state.focused} // PropTypes.bool
          onFocusChange={({ focused }) => this.setState({ focused })} // PropTypes.func.isRequired
        />
        <SingleDatePicker
          date={this.state.dateTwo} // momentPropTypes.momentObj or null
          onDateChange={dateTwo => this.setState({ dateTwo })} // PropTypes.func.isRequired
          focused={this.state.focusedTwo} // PropTypes.bool
          onFocusChange={({ focus }) => this.setState({ focus })} // PropTypes.func.isRequired
        />
      </form>
    )
  }
}
```

**Comment by ljharb** [comment]
Your first example consistently uses pairs of "focused" and "date".

Your second example uses a pair of "dateTwo", but then uses "focusedTwo" and "focus".

**Comment by nvnvenki** [comment]
I had the exact same problem. Had two single date pickers in the same container.
Below was the change made to fix the issue. Not sure how can this cause an issue.

![image](https://user-images.githubusercontent.com/1863449/49337131-62231a00-f634-11e8-8908-9daf6fa3f50a.png)

**Comment by Buddhimal** [comment]
> Your first example consistently uses pairs of "focused" and "date".
> 
> Your second example uses a pair of "dateTwo", but then uses "focusedTwo" and "focus".


This Doesn't work for me. What is the issue?


 constructor(props) {
        super(props);

        this.state = {
            fields: {},
            errors: {},
            date: '',
            date1: '',
            startDate: moment(),
            minDate: '1900/1/1',
            maxDate:moment(),
            endDate: moment(),
            dateRangeFocusedInput: true,
            focusedTwo: true,
            defaultSwitch: false,
            plainSwitch: false,
            iconSwitch: true,
            status: 'Active',
            statusClass:'text-primary'
        }
    }


<div className="form-group">
                                    <label className="col-sm-3 control-label">Date Of Birth*</label>
                                    <div className="col-sm-6">


                                        <SingleDatePicker
                                            date={date}
                                            onDateChange={date => this.setState({ date })}
                                            focused={this.state.focused}
                                            onFocusChange={({ focused }) => this.setState({ focused })}
                                            isOutsideRange={date => date.isBefore(minDate) || date.isAfter(maxDate)}
                                            
                                        />
                                        <span className="validation-error">{this.state.errors["dob"]}</span>
                                    </div>
                                </div>

                                <div className="form-group">
                                    <label className="col-sm-3 control-label">Joined Date*</label>
                                    <div className="col-sm-6">
                                          <SingleDatePicker
                                            date={date1}
                                            onDateChange={date1 => this.setState({ date1 })}
                                            focused={this.state.focusedTwo}
                                            onFocusChange={({ focusedTwo }) => this.setState({ focusedTwo })}
                                        />
                                        <span className="validation-error">{this.state.errors["joinDate"]}</span>
                                    </div>
                                </div>

**Comment by AndreaMorone** [comment]
Hi, 
you should do:
`onFocusChange={({ focused: focusedTwo }) => this.setState({ focusedTwo })}
`
because the onFocusChange callback passed back an object with a focused key, not a focusedTwo key.

**Comment by Zitle-Nancy** [comment]
Hi @AndreaMorone , for me this solution was very good and fixed my problem. 
Here is my code.

`<SingleDatePicker
                      placeholder="Salida"
                      numberOfMonths={1}
                      displayFormat="DD MMM YYYY"
                      date={this.state.dateStart} // momentPropTypes.momentObj or null
                      onDateChange={dateStart => this.setState({ dateStart })}
                      focused={this.state.focused} // PropTypes.bool
                      onFocusChange={({ focused }) => this.setState({ focused })}
                      id="departure" // PropTypes.string.isRequired,
                    />
                  </ContainerContact>
                  <ContainerContact xs={6} sm={6}>
                    <SingleDatePicker
                      numberOfMonths={1}
                      displayFormat="DD MMM YYYY"
                      placeholder="Regreso"
                      date={this.state.dateEnd} // momentPropTypes.momentObj or null
                      onDateChange={dateEnd => this.setState({ dateEnd })}
                      focused={this.state.focusedDateEnd} // PropTypes.bool
                      onFocusChange={({ focused: focusedDateEnd}) => this.setState({ focusedDateEnd })}
                      id="back" // PropTypes.string.isRequired,
                    />`
